code/game/objects/structures/mirror.dm

```diff
@@ -61,6 +65,19 @@
 	if(I.damtype == STAMINA)
 		return
 	if(shattered)
+		if(istype(I, /obj/item/weapon/weldingtool))
+			var/obj/item/weapon/weldingtool/WT = I
+			if(WT.remove_fuel(0, user))
+				user << "<span class='notice'>You begin repairing [src]...</span>"
+				playsound(src, 'sound/items/Welder.ogg', 100, 1)
+				if(do_after(user, 10, target = src))
+					if(!user || !WT || !WT.isOn())
+						return
+					user << "<span class='notice'>You repair [src].</span>"
+					shattered = 0
+					icon_state = initial(icon_state)
+					desc = initial(desc)
+					return

```

Comment:
This return needs to have one less indentation, otherwise starting to repair the mirror and moving away during the wait will play the hit_on_shattered_glass sound below.